### PR TITLE
Improve parameter inference

### LINKED ISSUE #680: [Doc]: Explain parameter inference
### Describe your documentation update

Add documentation on the parameter guessing/inferences. Also, perhaps change "guessing" to "inferring" in the code, per https://github.com/E3SM-Project/zppy/pull/672#issuecomment-2634772699

### PR DESCRIPTION
**Pull request by forsyth2**
## Summary

Objectives:
- Refactor parameter inference code to be more easily understood.
- Document parameter inference.

Issue resolution:
- Closes #680

Select one: This pull request is...
- [ ] a bug fix: increment the patch version
- [x] a small improvement: increment the minor version
- [ ] a new feature: increment the minor version
- [ ] an incompatible (non-backwards compatible) API change: increment the major version

## Big Change

- [ ] To merge, I will use "Create a merge commit". That is, this change is large enough to require multiple units of work (i.e., it should be multiple commits).

### 1. Does this do what we want it to do?

Required:
- [x] Product Management: I have confirmed with the stakeholders that the objectives above are correct and complete.
- [ ] Testing: I have added or modified at least one "min-case" configuration file to test this change. Every objective above is represented in at least one cfg.
- [x] Testing: I have considered likely and/or severe edge cases and have included them in testing.

If applicable:
- [ ] Testing: this pull request introduces an important feature or bug fix that we _must_ test often. I have updated the weekly-test configuration files, not just a "min-case" one.
- [ ] Testing: this pull request adds at least one new possible parameter to the cfg. I have tested using this parameter with and without any other parameter that may interact with it.

### 2. Are the implementation details accurate & efficient?

Required:
- [x] Logic: I have visually inspected the entire pull request myself.
- [x] Logic: I have left GitHub comments highlighting important pieces of code logic. I have had these code blocks reviewed by at least one other team member.

If applicable:
- [ ] Dependencies: This pull request introduces a new dependency. I have discussed this requirement with at least one other team member. The dependency is noted in `zppy/conda`, not just an `import` statement.

### 3. Is this well documented?

Required:
- [x] Documentation: by looking at the docs, a new user could easily understand the functionality introduced by this pull request.

### 4. Is this code clean?

Required:
- [x] Readability: The code is as simple as possible and well-commented, such that a new team member could understand what's happening.
- [x] Pre-commit checks: All the pre-commits checks have passed.

If applicable:
- [x] Software architecture: I have discussed relevant trade-offs in design decisions with at least one other team member. It is unlikely that this pull request will increase tech debt.

### CODE REVIEW

File: docs/source/parameters.rst [hunk 1]
```diff
@@ -15,3 +15,80 @@ See `this release's parameter defaults <https://github.com/E3SM-Project/zppy/blo
...
+
+* If ``ilamb_obs`` (the path to observation data for ``ilamb``) is undefined, assume it is the ``diagnostics_base_path`` from Mache plus ``/ilamb_data``.
+
+**Required parameters, by e3sm_diags set**
+
+TODO: Insert main table from Confluence
+
+These parameters are required for all model-vs-model runs: ``diff_title``, ``ref_name``, ``short_ref_name``. 
+
+TODO: Insert model-vs-model table from Confluence
```
Comment (forsyth2) [inline docs/source/parameters.rst:74]: https://acme-climate.atlassian.net/wiki/spaces/IPD/pages/4984209586/zppy+parameters+for+e3sm_diags

It's really difficult to create tables in `rst`, so for a big table, Confluence seemed like the better fit. For actually including it in the docs, I suppose we could add in screenshots?

**Review by forsyth2 — COMMENTED** [review]
@chengzhuzhang Let me know if https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.forsyth2/issue-680/parameters.html helps explain the parameter inference process a little more.

Note to self, to put docs up online:
```bash
cd docs
make html
mv _build/html /lcrc/group/e3sm/public_html/diagnostic_output/ac.forsyth2/issue-680
```

**Review by chengzhuzhang — APPROVED** [review]
Thanks to adding the documentation. The code change seems to be mostly name changes for functions. I'm running my zppy cfg with most of the tasks, will update if there are any issues.

**Comment by forsyth2** [comment]
Note to self, re: amending earlier commits

After https://github.com/E3SM-Project/zppy/pull/672 merged, I addressed merge conflicts on this branch. However, that didn't remove all the instances of the previous `guess` terminology. A few things needed to be updated manually. To do that, 1. I made the necessary changes, 2. I made a commit for those changes, 3. On `git rebase -i`, I simply reordered the commits to pick the refactor commit, fixup this new commit into that, and then pick the documentation commit.
